src/status_im2/subs/communities.cljs

```diff
@@ -219,18 +219,23 @@
 (re-frame/reg-sub
  :communities/categorized-channels
  (fn [[_ community-id]]
-   [(re-frame/subscribe [:communities/community community-id])])
- (fn [[{:keys [joined categories chats]}] [_ _]]
+   [(re-frame/subscribe [:communities/community community-id])
+    (re-frame/subscribe [:chats/chats])])
+ (fn [[{:keys [joined categories chats]} full-chats-data] [_ community-id]]
    (reduce
     (fn [acc [_ {:keys [name categoryID id emoji can-post?]}]]
-      (let [category (keyword (get-in categories [categoryID :name] (i18n/label :t/none)))]
+      (let [category                          (keyword (get-in categories
+                                                               [categoryID :name]
+                                                               (i18n/label :t/none)))
+            {:keys [unviewed-messages-count]} (get full-chats-data (str community-id id))]
         (update acc
                 category
                 #(vec (conj %1 %2))
-                {:name    name
-                 :emoji   emoji
-                 :locked? (or (not joined) (not can-post?))
-                 :id      id})))
+                {:name             name
+                 :emoji            emoji
+                 :unread-messages? (pos? unviewed-messages-count)
+                 :locked?          (or (not joined) (not can-post?))
```

Comment:
hmm, locked? is in reference to token-gating. Maybe we just change this to `false` and add a TODO with a github issue to immplement this with token gating? wdyt?

Reply:
`locked?` is for showing that small icon that informs that channel can't be open. And as far as I understand, it is not only about token gating. It is also locked when you did not join the community

Reply:
I think we should double check that with designers. I would assume you can not join any locked channel if you are not a member of a community.

Reply:
And it would be locked when you have not joined either way because the channel is locked by some token permissions. I could have it wrong however, but let's sync with designers on this

Reply:
Yeah, good idea! will write this down to ask on a next syncup!

Reply:
Cool 🤙